osu.Game/Screens/Select/Carousel/CarouselBeatmap.cs

```diff
@@ -26,6 +26,12 @@ public override void Filter(FilterCriteria criteria)
 
             bool match = criteria.Ruleset == null || Beatmap.RulesetID == criteria.Ruleset.ID || Beatmap.RulesetID == 0 && criteria.Ruleset.ID > 0 && criteria.AllowConvertedBeatmaps;
 
+            if(criteria.DisplayStarsMinimum.HasValue)
+                match &= Beatmap.StarDifficulty >= criteria.DisplayStarsMinimum;
+
+            if (criteria.DisplayStarsMaximum.HasValue)
```

Comment:
Is there a case where these don't have a value? Could they be non-nullable (because the config has a set, non-null value)?

Reply:
In tests yes. Not quite sure about the game but I don't think they could be. I guess then I can remove the `if` and ensure existing values in tests to not break them. I'll check once I'm home again.

Reply:
Yeah I think that'd be better, just to remove the special case specific to tests.